util/types/datum.go

```
@@ -1630,10 +1591,9 @@ func CoerceDatum(sc *variable.StatementContext, a, b Datum) (x, y Datum, err err
 			y.SetFloat64(float64(y.GetInt64()))
 		case KindUint64:
 			y.SetFloat64(float64(y.GetUint64()))
-		case KindMysqlHex:
-			y.SetFloat64(y.GetMysqlHex().ToNumber())
-		case KindMysqlBit:
-			y.SetFloat64(y.GetMysqlBit().ToNumber())
+		case KindBinaryLiteral, KindMysqlBit:
+			val, _ := y.GetBinaryLiteral().ToInt()
```

error is not handled.